framework/include/utils/MooseTypes.h

```diff
@@ -166,6 +166,8 @@ enum KernelType
 {
   KT_TIME = 0,
   KT_NONTIME = 1,
+  KT_NONEIGEN = 2,
+  KT_EIGEN  = 3,
```

Comment:
Do we use any of the enumeration constants directly? Maybe we should just drop them.